### ISSUE TITLE
Akka.Persistence.Sql.Common: DeleteMessages when journal is empty causes duplicate key SQL exception

### ISSUE THREAD

**Issue by Aaronontheweb** [akkadotnet/akka.net#3721]
Version: Akka.NET v1.3.11

Use case: when using an `AtLeastOnceDeliveryActor`, I'm not using the journal at all - but I do save snapshots of my `AtLeastOnceDelivery` state regularly. 

When I receive a `SaveSnapshotSuccess` message, I call:

```
Command<SaveSnapshotSuccess>(saved =>
            {
                var seqNo = saved.Metadata.SequenceNr;
                DeleteMessages(seqNo);
                DeleteSnapshots(new SnapshotSelectionCriteria(seqNo,
                    saved.Metadata.Timestamp.AddMilliseconds(-1))); // delete all but the most current snapshot
            });
```

Out of a force of habit from my other actors that use Akka.Persistence.

The `DeleteMessages` call in this case causes a `SqlException` to be raised in the `Metadata` table used by the Akka.Persistence.Sql.Common driver, due to this line in the driver:

https://github.com/akkadotnet/akka.net/blob/666811cb5868847c2c51480ae1136a3d912cdd0f/src/contrib/persistence/Akka.Persistence.Sql.Common/Journal/QueryExecutor.cs#L617-L629

Since the sequence numbers are both 0, the driver tries to insert another `(persistenceId, 0)` record into the Metadata table even though one already exists.

I'll send a repo along for this shortly.

**Comment by ismaelhamed** [comment]
GH-3468

**Comment by Aaronontheweb** [comment]
@ismaelhamed good to know - I'll go back and review that PR again

**Comment by Aaronontheweb** [comment]
closed via #3468
